app/controllers/incomeSources/add/AddIncomeSourceController.scala

```diff
@@ -62,38 +66,40 @@ class AddIncomeSourceController @Inject()(val addIncomeSources: AddIncomeSources
         getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
           implicit mtdItUser =>
             handleRequest(
-              sources = mtdItUser.incomeSources,
               isAgent = true,
+              homePageCall = homePageCallAgent,
+              sources = mtdItUser.incomeSources,
               backUrl = controllers.routes.HomeController.showAgent.url
             )
         }
   }
 
-  def handleRequest(sources: IncomeSourceDetailsModel, isAgent: Boolean, backUrl: String)
+  def handleRequest(sources: IncomeSourceDetailsModel,
+                    homePageCall: Call,
+                    isAgent: Boolean,
+                    backUrl: String)
                    (implicit user: MtdItUser[_]): Future[Result] = {
-    Future.successful(
-      if (isDisabled(IncomeSources)) {
-        Redirect(controllers.routes.HomeController.show())
-      } else {
-        incomeSourceDetailsService.getAddIncomeSourceViewModel(sources) match {
-          case Right(viewModel) =>
-            Ok(addIncomeSources(
-              viewModel,
-              isAgent = isAgent,
-              backUrl = backUrl
-            ))
-          case Left(ex) =>
-            if (isAgent) {
-              Logger("application").error(
-                s"[Agent][AddIncomeSourceController][handleRequest] - Error: ${ex.getMessage}")
-              itvcErrorHandlerAgent.showInternalServerError()
-            } else {
-              Logger("application").error(
-                s"[AddIncomeSourceController][handleRequest] - Error: ${ex.getMessage}")
-              itvcErrorHandler.showInternalServerError()
-            }
-        }
+    if (isDisabled(IncomeSources)) {
+      Future.successful(Redirect(homePageCall))
+    } else {
+      incomeSourceDetailsService.getAddIncomeSourceViewModel(sources) match {
```

Comment:
would be easier to wrap this whole expression with Future, but I guess we can leave things as is